data-runtime/src/main/java/io/micronaut/data/runtime/criteria/RuntimePersistentEntityCriteriaUpdate.java

```diff
@@ -73,14 +73,15 @@ protected void setValue(String attributeName, Object value) {
 
     @NotNull
     private ParameterExpression<?> asParameter(Object exp) {
-        if (exp instanceof ParameterExpression) {
-            return (ParameterExpression<?>) exp;
+        if (exp instanceof ParameterExpression<?> parameterExpression) {
+            return parameterExpression;
         }
-        Objects.requireNonNull(exp);
         Class<Object> type;
         Object value;
-        if (exp instanceof LiteralExpression) {
-            LiteralExpression literalExpression = (LiteralExpression<?>) exp;
+        if (exp == null) {
```

Comment:
The correct might be to use `CriteriaBuilder#nullLiteral` can you please check how does Hibernate handels it.

Reply:
Yes, changed it to use criteriaBuilder.nullLiteral and added back Objects.requireNonNull